TSOClient/FSO.Server.Api.Core/Controllers/UserOAuthController.cs

```diff
@@ -0,0 +1,128 @@
+﻿using FSO.Server.Api.Core.Utils;
+using FSO.Server.Common;
+using FSO.Server.Servers.Api.JsonWebToken;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Web;
+using System.Web.Http;
+
+namespace FSO.Server.Api.Core.Controllers
+{
+    [Route("userapi/oauth/token")]
+    [ApiController]
+    public class UserOAuthController : ControllerBase
```

Comment:
VERY IMPORTANT: make sure this auth does not allow admin actions to be performed. (admin api) Since it doesn't add the "admin" or "moderator" claims, it will probably be fine, but it definitely would not hurt to check.